pkgs/development/python-modules/pytask/default.nix

```
@@ -0,0 +1,87 @@
+{
+  lib,
+  buildPythonPackage,
+  pythonOlder,
+  fetchFromGitHub,
+  hatchling,
+  hatch-vcs,
+  attrs,
+  click,
+  click-default-group,
+  networkx,
+  optree,
+  packaging,
+  pluggy,
+  rich,
+  sqlalchemy,
+  universal-pathlib,
+  pytestCheckHook,
+  deepdiff,
+  nbmake,
+  pexpect,
+  pytest-cov,
+  pytest-xdist,
+  syrupy,
+  aiohttp,
+  git,
+}:
+buildPythonPackage rec {
+  pname = "pytask";
+  version = "0.5.1";
+  format = "pyproject";
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "pytask-dev";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-b+sS+l0Rp5bb8Dh6UBv3xHYTYKFp3dD5AuLqxB3n6Go=";
+  };
+
+  nativeBuildInputs = [
+    hatchling
+    hatch-vcs
+  ];
+
+  propagatedBuildInputs = [
+    attrs
+    click
+    click-default-group
+    networkx
+    optree
+    packaging
+    pluggy
+    rich
+    sqlalchemy
+    universal-pathlib
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    git
+    deepdiff
+    nbmake
+    pexpect
+    pytest-cov
+    pytest-xdist
+    syrupy
+    aiohttp
+  ];
+
+  # The test suite runs the installed command for e2e tests
+  preCheck = ''
+    export PATH="$PATH:$out/bin";
+  '';
+
+  disabledTests = [
+    # This accesses the network
+    "test_download_file"
+  ];
+
+  meta = with lib; {
+    description = "Workflow management system that facilitates reproducible data analyses";
+    homepage = "https://github.com/pytask-dev/pytask";
```

```suggestion
    homepage = "https://github.com/pytask-dev/pytask";
    changelog = "https://github.com/pytask-dev/pytask/releases/tag/v${version}";
```
Adding `meta.changelog` is useful for future updates.